### PR TITLE
chore: improve output of `-h/--help`

### PR DESCRIPTION
**Pull request by onbjerg**
## Motivation

The output of `-h` and `--help` is inconsistent and sometimes not super helpful.

## Solution

Goes through all help text and updates the wording. I also move some of the longer-form help text into `long_help` to shorten the output of `-h`.

I'm going through this as I am rewriting the command references in the book.

**Tasks**

- [x] Go through the rest of Cast
- [x] Go through the rest of Forge
- [x] Be consistent with the help text of function signatures and block tags
- [x] Note down commands that are good candidates for merging
- [x] Improve docs on `Wallet` struct
- [x] Remove unused options in each command (deferred to #1274)
- [x] Link to the book at the end of `-h`/`--help`

### CODE REVIEW

**Comment by onbjerg** [comment]
Current status on Forge reference:

Easy

- [ ] forge install
- [ ] forge update
- [ ] forge remove
- [ ] forge init
- [ ] forge clean
- [ ] forge flatten
- [ ] forge verify-contract
- [ ] forge verify-check
- [ ] forge bind
- [ ] forge remappings

Hard

- [ ] forge run
- [ ] forge create
- [ ] forge tree
- [ ] forge config
- [ ] forge test
- [ ] forge snapshot

File: cli/src/cmd/forge/build.rs [hunk 1]
```diff
@@ -48,169 +48,201 @@ impl<'a> From<&'a BuildArgs> for Figment {
...
-        help = "force recompilation of the project, deletes the cache and artifacts folders",
+        help_heading = "COMPILER OPTIONS",
+        help = "Use the Yul intermediate representation compilation pipeline.",
         long
     )]
     #[serde(skip)]
-    pub force: bool,
+    pub via_ir: bool,
+
+    #[clap(flatten, next_help_heading = "PROJECT OPTIONS")]
```
Comment (mattsse) [inline cli/src/cmd/forge/build.rs:119]: maybe PROJECT LAYOUT/DIR OPTIONS ?

**Review by mattsse — APPROVED** [review]
lgtm,

like the breakdown of buildargs

File: cli/src/cmd/forge/build.rs (hunk 1 above)
Comment (onbjerg) [reply cli/src/cmd/forge/build.rs:119]: Can do in a follow up PR

**Pull request merged**